scripts/validate-artifacts.ps1

```diff
@@ -34,138 +34,155 @@ Write-Output "Validating artifacts"
+    Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Unable to verify the subnet name. Please make sure that your script creates a subnet called 'default' and try again."
+    throw "Virtual Machine uses wrong OS image. Please re-deploy VM using Ubuntu Server 22.04 and try again" 
 }
 
-$sshKey = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Compute/sshPublicKeys")
-if ($sshKey) {
-    if ($sshKey.name.Count -eq 1) { 
-        Write-Output "`u{2705} Checked if the public SSH key resource exists - OK"
-    }  else { 
-        Write-Output `u{1F914}
-        throw "More than one public SSH key resource was found in the VM resource group. Please make sure that your script creates only one public SSH key resource and try again."
-    }
-} else {
+if ($virtualMachine.properties.hardwareProfile.vmSize -eq "Standard_B1s") { 
+    Write-Output "`u{2705} Checked Virtual Machine size - OK"
+} else { 
     Write-Output `u{1F914}
-    throw "Unable to find public SSH key resouce. Please make sure that your script creates a public SSH key resouce and try again."
+    throw "Virtual Machine size is not set to B1s. Please re-deploy VM with size set to B1s and try again."
 }
 
-$sshKeyName = $sshKey.name.Replace("[parameters('sshPublicKeys_", "").Replace("_name')]", "")
-if ($sshKeyName -eq "linuxboxsshkey") { 
-    Write-Output "`u{2705} Checked the public ssh key name - OK"
+if ($virtualMachine.properties.osProfile.linuxConfiguration.disablePasswordAuthentication -eq $true) { 
+    Write-Output "`u{2705} Checked Virtual Machine OS user authentification settings - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Unable to verify the public ssh key name. Please make sure that your script creates a public ssh key called 'linuxboxsshkey' and try again."
+    throw "Virtual Machine uses password authentification. Please re-deploy VM using SSH key authentification for the OS admin user and try again. "
 }
 
-$virtualMachines = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Compute/virtualMachines" )
-if ($virtualMachines) {
-    if ($virtualMachines.name.Count -eq 2) { 
-        Write-Output "`u{2705} Checked if Virtual Machines exists - OK."
+
+$pip = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/publicIPAddresses")
+if ($pip) {
+    if ($pip.name.Count -eq 1) { 
+        Write-Output "`u{2705} Checked if the Public IP resource exists - OK"
     }  else { 
         Write-Output `u{1F914}
-        throw "Wrong number of Virtual Machine resource was found in the task resource group. Please make sure that your script creates 2 VMs (each in own availability zone) and try again."
+        throw "More than one Public IP resource was found in the VM resource group. Please delete all un-used Public IP address resources and try again."
```

Comment:
There is a typo in the error message: 'resouce' should be 'resource'. Correcting this will make the error message clearer.